nj2k/src/org/jetbrains/kotlin/nj2k/tree/statements.kt

```diff
@@ -84,10 +84,22 @@ class JKKtWhenStatement(
 
 class JKKtConvertedFromForLoopSyntheticWhileStatement(
     variableDeclaration: JKStatement,
-    whileStatement: JKWhileStatement
-) : JKStatement() {
+    whileStatement: JKWhileStatement,
+    forLoopUpdaters: List<JKStatement>
+) : JKLoopStatement() {
     var variableDeclaration: JKStatement by child(variableDeclaration)
     var whileStatement: JKWhileStatement by child(whileStatement)
+    // For the use in ContinueStatementConverter
+    var forLoopUpdaters: List<JKStatement> by children(forLoopUpdaters)
+
+    // No child check because the parent of whileStatement.body is whileStatement
+    // and check will fail.
+    override var body: JKStatement
+        get() = whileStatement.body
+        set(v) {
+            whileStatement.body = v
+        }
```

Comment:
You can just move all needed stuff from `JKWhileStatement` to `JKKtConvertedFromForLoopSyntheticWhileStatement` and get rid of that delegation